### PR TITLE
Allow for parameters in SHORTEST

### PR DESCRIPTION
**Pull request by gem-neo4j**

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-1124.surge.sh

Updated pages:
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-1124.surge.sh/cypher-manual/25/deprecations-additions-removals-compatibility)
- [Syntax and semantics](https://neo4j-docs-cypher-1124.surge.sh/cypher-manual/25/patterns/reference)
- [Shortest paths](https://neo4j-docs-cypher-1124.surge.sh/cypher-manual/25/patterns/shortest-paths)

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc [hunk 1]
```diff
@@ -93,6 +93,26 @@ MATCH (n)-[r:REL]->(m) SET n = properties(r)
...
+----
+MATCH p = ANY $param (:A)-[:R]->{0,10}(:B)
+----
+
+[source, cypher, role=noheader]
+----
+MATCH SHORTEST $param GROUPS (:A)-[:R]->{0,10}(:B)
+----
+
+a| Introduced the allowance of parameters to the xref:patterns/shortest-paths.adoc[SHORTEST and ANY path patterns].
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]:
```suggestion
a| Parameters can now be used in xref:patterns/shortest-paths.adoc[`SHORTEST` and `ANY` path patterns].
```

File: modules/ROOT/pages/patterns/shortest-paths.adoc [hunk 2]
```diff
@@ -64,7 +64,7 @@ CREATE (asc)-[:LINK {distance: 7.25}]->(cnm),
   (wof)-[:LINK {distance: 0.65}]->(wos)
 ----
 
-The paths matched by a xref:patterns/fixed-length-patterns.adoc#path-patterns[path pattern] can be restricted to only the shortest (by number of hops) by including the keyword `SHORTEST k`, where `k` is the number of paths to match.
+The paths matched by a xref:patterns/fixed-length-patterns.adoc#path-patterns[path pattern] can be restricted to only the shortest (by number of hops) by including the keyword `SHORTEST k`, where `k` is the number of paths to match, and can be either an `INTEGER` literal or a parameter which resolves to an `INTEGER`.
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/patterns/shortest-paths.adoc]:
```suggestion
The paths matched by a xref:patterns/fixed-length-patterns.adoc#path-patterns[path pattern] can be restricted to only the shortest (by number of hops) by including the keyword `SHORTEST k`, where `k` is the number of paths to match, and can be either an `INTEGER` literal or, as of Neo4j 2025.01, a parameter which resolves to an `INTEGER`.
```

**Review by JPryce-Aklundh — APPROVED** [review]
Just 2 small suggestions :)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-1124.surge.sh

Updated pages:
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-1124.surge.sh/cypher-manual/25/deprecations-additions-removals-compatibility)
- [Syntax and semantics](https://neo4j-docs-cypher-1124.surge.sh/cypher-manual/25/patterns/reference)
- [Shortest paths](https://neo4j-docs-cypher-1124.surge.sh/cypher-manual/25/patterns/shortest-paths)

**Review by JoelBergstrand — APPROVED** [review]
No comments from me!

**Comment by neo-technology-commit-status-publisher** [comment]
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.

**Pull request merged**